server/plugins/messageStorage/sqlite.ts

```diff
@@ -243,31 +227,59 @@ class SqliteMessageStorage implements ISqliteMessageStorage {
 		query.offset = parseInt(query.offset as string, 10) || 0;
 		params.push(String(query.offset));
 
+		const rows = await this.serialize_fetchall(select, ...params);
+		const response: SearchResponse = {
+			searchTerm: query.searchTerm,
+			target: query.channelName,
+			networkUuid: query.networkUuid,
+			offset: query.offset as number,
+			results: parseSearchRowsToMessages(query.offset as number, rows).reverse(),
+		};
+
+		return response;
+	}
+
+	canProvideMessages() {
+		return this.isEnabled;
+	}
+
+	private run(stmt: string, ...params: any[]) {
+		this.serialize_run(stmt, params).catch((err) =>
+			log.error(`failed to run ${stmt}`, String(err))
+		);
+	}
+
```

Comment:
Might as well embrace the `async` way fully:

```typescript
private async run(stmt: string, ...params: any[]) {
    try {
        await this.serialize_run(stmt, params)
    } catch (err) {
        log.error(`failed to run ${stmt}`, String(err));
    );
}
```

Reply:
yes. That went the way of the Dodo since... just didn't push the new changes yet.